File: Hard/23._Merge_k_Sorted_Lists/solution.py
# Beats 96.95% Runtime O(N∗Log(N)) Complexity
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        final = []
        for node in lists:
            cur = node
            while cur is not None:
                final.append(cur.val)
                cur = cur.next
        if len(final) == 0:
            return
        final.sort()
        cur=ListNode()
        root = cur
        for i in range(0,len(final)):
            cur.val = final[i]
            cur.next = ListNode() if i != len(final)-1 else None
            cur = cur.next
        return root

File: Hard/23._Merge_k_Sorted_Lists/solution.go
// beats 71.25% runtime 
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 import (
    "slices"
)

func mergeKLists(lists []*ListNode) *ListNode {
    final := []int{}
    for i := range len(lists) {
        node := lists[i]
        cur := node
        for cur != nil {
            final = append(final, cur.Val)
            cur = cur.Next
        }

    }
    if len(final) == 0 {
        return nil
    }
    slices.Sort(final)
    cur := &ListNode{}
    root := cur
    for i := range len(final) {
        cur.Val = final[i]
        if i < len(final)-1 {
            cur.Next = &ListNode{}
            cur = cur.Next
        }
    }
    return root
}
